lib/parsers/csv_layout/csv_parser.rb

```diff
@@ -17,19 +17,18 @@ class CsvParser
       attr_reader :data, :parsed, :parsed_changes, :components, :line_parser
 
       DEFAULT_COMPONENTS = {
-          barcode_parser: Parsers::CsvLayout::BarcodeParser,
-          location_parser: Parsers::CsvLayout::LocationParser,
-          location_validator: Parsers::CsvLayout::Validators::LocationValidator,
-          barcode_validator: Parsers::CsvLayout::Validators::FluidxBarcodeValidator,
-          line_parser: Parsers::CsvLayout::LineParser,
-          line_reader: Parsers::CsvLayout::LineReader
+        barcode_parser: Parsers::CsvLayout::BarcodeParser,
+        location_parser: Parsers::CsvLayout::LocationParser,
+        location_validator: Parsers::CsvLayout::Validators::LocationValidator,
+        barcode_validator: Parsers::CsvLayout::Validators::FluidxBarcodeValidator,
+        line_parser: Parsers::CsvLayout::LineParser,
+        line_reader: Parsers::CsvLayout::LineReader
       }
 
       def initialize(str, component_defs = {})
         @parsed = false
         @input = str
         @components = self.class::DEFAULT_COMPONENTS.merge(component_defs)
-        valid?
```

Comment:
Valid in this context is quite slow, and doesn't really do much. While it does populate the error_messages, this gets blown away next time we call valid?, and in no case (unless I managed to miss one) do we try to check the error messages without calling valid? immediately previously/